src/MyRankings.jsx

```diff
@@ -1,6 +1,68 @@
-import React from "react";
-// import "./Recommendations.css";?
-function MyRankings() {
-  return <h1>My Rankings</h1>;
+import React, { useState, useEffect } from "react";
+import "./MyRankings.css";
+import FlixterHeader from "./FlixterHeader";
+import { getRatingColor } from "./utils";
+
+function MyRankings({ movies }) {
+  const [page, setPage] = useState(1);
+  const [data, setData] = useState();
+
+  useEffect(() => {
+    const options = {
+      method: "GET",
+      headers: {
+        accept: "application/json",
+        Authorization: `Bearer ${import.meta.env.VITE_API_KEY}`,
+      },
+    };
+    fetch(
+      `https://api.themoviedb.org/3/movie/now_playing?language=en-US&page=`,
+      options
+    )
+      .then((response) => response.json())
+      .then((newData) => {
+        setData((data) => ({
+          ...newData,
+          results:
+            page != 1
+              ? [...data.results, ...newData.results]
+              : [...newData.results],
+        }));
+      })
+      .catch((error) => console.error("Error fetching data:", error));
+  }, [page]);
+
+  return (
+    <div className="rankingsPage">
+      <FlixterHeader></FlixterHeader>
+      <div className="rankingsTitle">WATCHED MOVIES</div>
+      <div className="sub">Explore personal ratings for movies you've seen</div>
+
+      <div id="rankingscontainer">
+        {data?.results
+          ?.slice(0, 20)
+          .sort((a, b) => b.vote_average - a.vote_average)
```

Comment:
The set of movies comes in certain order defined by the service (themoviedb), then you take the first 20 and sort them by vote average. This is fine as long as only the N (20 in this case) first movies are going to be displayed always. 

If you plan on doing pagination, then the solution will need to be different. Either they come sorted the way you need them to be, or you sort all of them in the client (not performant)

Reply:
The Movies Db movies are just a placeholder, the movies will be coming in through a database fetch, but I think the same question still applies. Pagination is my plan i will have to change the sorting accordingly